machine/v1/types_awsplacementgroup.go

```diff
@@ -0,0 +1,163 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AWSPlacementGroup ensures that a placement group matching the given configuration exists within AWS
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".spec.groupType",description="Placement Group Type"
+// +kubebuilder:printcolumn:name="Management",type="string",JSONPath=".spec.managementState",description="Management State"
+// +kubebuilder:printcolumn:name="Replicas",type="integer",JSONPath=".status.replicas",description="EC2 Replicas within the Placement Group"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="AWSPlacementGroup age"
+type AWSPlacementGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AWSPlacementGroupSpec   `json:"spec,omitempty"`
+	Status AWSPlacementGroupStatus `json:"status,omitempty"`
+}
+
+type AWSPlacementGroupSpec struct {
+	// AWSPlacementGroupManagementSpec defines the configuration for a managed or unmanaged placement group.
+	AWSPlacementGroupManagementSpec `json:",inline"`
```

Comment:
you cannot have this inline because the `credentialsSecret` isn't controlled by the the value of the managementstate.

Reply:
I thought this was ok, there's an example doing the same (`TopLevelUnion`) in the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md#go-tags), do you know what this would break/why this wouldn't work? Is it a UX thing rather than a technical issue?